### ISSUE TITLE
Implement a Thread Limit Handler

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#844]
Jetty has a DoSFilter that can be deployed to give reasonable defence against DoS attack, at least as best you can do within a server.

However, DoSFilter has lots of configuration options, not all of them useful, and as a filter can be hard to deploy as a post process.    Thus it may be a better alternative to create a DoSHandler that is slightly simplified and improved, that can easily be applied as a module.

The DoSFilter has support for tracking requests within a session.  This can be useful against accidental DoS attacks (Eg javascript gone crazy), but any malicious DoS attack will not use the same session cookie.  Essentially sessions cannot be trusted.

The information that can be trusted includes:
- Remote IP can be directly determined, but this will often be a proxy and/or NAT router rather than the ultimate client.  So it is a poor measure for grouping HTTP/1 connection to a single source, but there are few other options for that.
- Forwarded-For headers, can be trusted in some circumstances.  If you are behind your own proxy that will add those headers, then they can be meaningful, but the standard ForwardedRequestCustomizer takes the left-most value as the original client - which makes it possible for the client to supply this data.   It is probably better to take the rightmost value, when it is known that you have a single proxy and to ignore if there are no known proxies.
- Remote IP + Remote Port can be directly determined and uniquely identify a client connection so long as any proxy does not aggregate traffic.    But we already have a local concept of connection, so this does not add too much value other than a connection id.    Limiting resources by connection is good for HTTP/2, which allows multiple outstanding requests, but of limited value for HTTP/1.   Currently it is a modal configuration in the DoSFilter, so it cannot be configured for both optimal HTTP/1 and HTTP/2.

Currently the DoSFilter apply a request rate check and only if the request rate is exceeded is a thread limit applied.     It may be simpler and almost as effective to just have a thread limit per connection and per remote IP.

In summary I am proposing:
- Create a DosHandler
- It will apply a per connection thread limit and a per remote client thread limit.
- remote client will be determined either by direct IP or right-most forwarded header
- module created to easily apply DosHandler to server
